src/sentry/static/sentry/app/views/organizationIncidents/details/activity.jsx

```diff
@@ -247,8 +243,58 @@ class ActivityContainer extends React.Component {
     }
   }
 
+  handleCreateNote = async note => {
+    const {api, params} = this.props;
+    const {incidentId, orgId} = params;
+
+    this.setState({
+      createBusy: true,
+    });
+
+    const newActivity = {
+      comment: note.text,
+      type: INCIDENT_ACTIVITY_TYPE.COMMENT,
+      dateCreated: new Date(),
+      user: ConfigStore.get('user'),
+      id: uniqueId(),
+      incidentIdentifier: incidentId,
+    };
+
+    this.setState(state => ({
+      createBusy: false,
+
+      activities: [newActivity, ...(state.activities || [])],
+    }));
+
+    try {
+      const newNote = await createIncidentNote(api, orgId, incidentId, note);
+
+      this.setState(state => {
+        const activities = [
+          newNote,
+          ...state.activities.filter(activity => activity === newActivity),
```

Comment:
This looks like it will keep only records that match the new one. Do you want to keep only those that don't match?

Reply:
Oops, good catch